### PR TITLE
Update windows installer

### PR DESCRIPTION
**Pull request by rolandlo**
- [x] Install plugins and resource folder
- [x] Generate Windows icon file xournalpp.ico automatically
- [x] Add Xournal++ to the "Open With" Context for PDF files (#2585)
- [x] Remove the launcher
- [x] Test everything carefully

### CODE REVIEW

**Comment by Febbe** [comment]
Can we get rid of the launcher? I would like to have only one executable.

**Comment by Technius** [comment]
The launcher should be safe to remove. It was needed previously because some paths were hardcoded into the application, but ever since the installation became relocatable, the launcher became unnecessary.

**Comment by Technius** [comment]
I found some time today to set up a Windows VM and change some of the Windows build settings. Would you like me to commit the changes to this PR directly or open a new one?

**Comment by rolandlo** [comment]
@Technius Yes, please submit your changes to this PR directly. Yes, the ImageMagick convert doesn't work yet.

**Comment by Technius** [comment]
See the commit message for details on what I did. I updated the build script to use the CMake install target as a base, so it just needs to copy the libraries and runtime dependencies. Tested on my Windows VM, but I could've missed some things when I copied my patch over to my Linux installation.

I think we can safely remove the launcher at this point.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -69,6 +69,28 @@ file (GLOB xournalpp_SOURCES_RECURSE
   pdf/popplerapi/*.cpp
 )
 
+if (WIN32)
```
Comment (Technius) [inline src/CMakeLists.txt:72]: This code for generating the icons could probably be combined into the one below.

File: windows-setup/package.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+## Windows Setup script
```
Comment (Technius) [inline windows-setup/package.sh:3]: I forgot to update this, will do it later.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -69,6 +69,28 @@ file (GLOB xournalpp_SOURCES_RECURSE
...
+  # generate .ico from .png
+  set (ICON_SIZES 16 32 48 256)
+  foreach (SIZE IN LISTS ICON_SIZES)
+    set (RESIZED_ICON ${PROJECT_BINARY_DIR}/xournalpp-icon-${SIZE}.png)
+    add_custom_command (
+	  OUTPUT ${RESIZED_ICON}
+	  COMMAND rsvg-convert ${PROJECT_SOURCE_DIR}/ui/pixmaps/com.github.xournalpp.xournalpp.svg -o ${RESIZED_ICON}
```
Comment (rolandlo) [inline src/CMakeLists.txt:79]: You should specify width and height here. Otherwise all icons are identic, 128x128 pixel.

**Comment by rolandlo** [comment]
I have tested the version from my last commit (without launcher and with the .ico file created from .png files with different resolutions) and it worked fine. The icon was generated correctly and Xournal++ ran normally, no problem with pressure sensitivity of the pen. Icons are displayed correctly.

File: src/CMakeLists.txt (hunk 1 above)
Comment (rolandlo) [reply src/CMakeLists.txt:72]: I addressed this.

File: src/CMakeLists.txt (hunk 3 above)
Comment (rolandlo) [reply src/CMakeLists.txt]: I fixed this.

**Comment by rolandlo** [comment]
The last commit (about opening pdf-files via the "Open With" context-menu) doesn't quite work as I hoped for. It must probably be reverted or at least modified.

**Comment by Technius** [comment]
Xournal++ is now an entry in the `Open With...` popup. See the commit description for the full changelog.

One thing we are missing is version information in the installer. We will need to insert the version info into the `src/win32/xpp.rc` file during the CMake configuration.

**Comment by Technius** [comment]
Last thing we need to do is test the installer.

@rolandlo Can you do a release test to ensure that the installer will build correctly?

**Comment by rolandlo** [comment]
@Technius Sure, I will do the testing.

**Comment by rolandlo** [comment]
Some comments from testing:
- The release pipeline now generates the installer corrrectly (I had to update the name of the script in `release.yml`)
- The version is correctly displayed in the installer
- The uninstaller works now
- The new icon is displayed everywhere except in the start menu, where I still get the old icon displayed
- Attaching to the the taskbar works fine
- "Open with" works now
- The folder in which Xournal++ is executed (via the start menu link) is wrong. It should be executed in the `bin` subfolder. Otherwise localizations are not detected and the language cannot be changed to anything different than English
- The folder structure has changed (as far as I can see). When installing with the new installer into the same folder, some old subfolders are left (in particular the `ui` folder is left in the `Xournal++` directory)
- The LaTeX settings panel does not open the right folder (at first) when attempting to change the template
- I have only tested "Install for all users"

**Comment by Technius** [comment]
> The new icon is displayed everywhere except in the start menu, where I still get the old icon displayed

I noticed this too, except I didn't get any icon at all. I haven't figured out if this is a problem due to the legacy install or if this is due to a missing registry key, etc.

**Comment by Technius** [comment]
I made some more fixes (see commit message for detailed changes). Installing on top of the legacy version should generally work now. I have not fixed the localization issue though, and would prefer putting it off into a follow-up PR since this issue is blocking a number of other PRs / potential fixes.

> The new icon is displayed everywhere except in the start menu, where I still get the old icon displayed

This has been fixed now. Part of the problem was due to the old start menu entries not being removed, another part was due to the `SHChangeNotify` issue I pointed out above.

> The LaTeX settings panel does not open the right folder (at first) when attempting to change the template

Not sure why this is happening.

> I have only tested "Install for all users"

We should probably disable or delete the "Install for current user" setting. I currently don't have the time/intention to support it.

**Comment by Technius** [comment]
@rolandlo Let's get this merged in, say, 48 hours. Do you have any remaining concerns / tests you want to do?

**Comment by rolandlo** [comment]
@Technius I have just tested it and the icon issue is fixed. Great! The LaTeX template path also worked. Maybe I had a bad path in my settings file before. I am fine with disabling or deleting the "Install for current user" setting.
I don't have any remaining concerns.

**Comment by rolandlo** [comment]
I have also checked once more that the PR builds successfully on Azure Pipelines. It looks all good.

**Comment by Technius** [comment]
I'll disable the "install for current user settings" tomorrow and merge this PR afterwards.

**Pull request merged**